tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_mappings.go

```diff
@@ -0,0 +1,189 @@
+package dataapigeneratorjson
+
+import (
+	"fmt"
+	"sort"
+
+	dataApiModels "github.com/hashicorp/pandora/tools/importer-rest-api-specs/components/dataapigeneratorjson/models"
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
+)
+
+func mapTerraformSchemaMappings(input resourcemanager.MappingDefinition) (*dataApiModels.TerraformMappingDefinition, error) {
+	output := dataApiModels.TerraformMappingDefinition{}
+
+	fieldMappings := make([]dataApiModels.TerraformFieldMappingDefinition, 0)
+	modelToModelMappings := make([]dataApiModels.TerraformModelToModelMappingDefinition, 0)
+	for _, item := range input.Fields {
+		switch item.Type {
+		case resourcemanager.DirectAssignmentMappingDefinitionType:
+			{
+				// DirectAssignment Mappings come solely from the Mapping themselves
+				fieldMappings = append(fieldMappings, dataApiModels.TerraformFieldMappingDefinition{
+					Type: dataApiModels.DirectAssignmentTerraformFieldMappingDefinitionType,
+					DirectAssignment: &dataApiModels.TerraformFieldMappingDirectAssignmentDefinition{
+						SchemaModelName: item.DirectAssignment.SchemaModelName,
+						SchemaFieldPath: item.DirectAssignment.SchemaFieldPath,
+						SdkModelName:    item.DirectAssignment.SdkModelName,
+						SdkFieldPath:    item.DirectAssignment.SdkFieldPath,
+					},
+				})
+				continue
+			}
+
+		case resourcemanager.ModelToModelMappingDefinitionType:
+			{
+				// ModelToModel mappings need to be output both for Fields and for the Models themselves
+				// this is because a ModelToModel mapping must exist from the Schema Model to the SDK Model
+				// but also from a given Schema Field to a given SDK Model.
+				//
+				// This allows both the mapping function between two models to be generated (the `ModelToModelMappings`)
+				// and the mapping between a given Schema Field and an SDK Model (so that we know to call the
+				// mapping function defined in the `ModelToModelMappings`).
```

Comment:
It's unclear to me what this comment is really trying to say or add in this context of a case/switch and is potentially causing more confusion.

Is it saying that regardless of what the schema looks like there always needs to be a `ModelToModel` present with values in it? or empty? 
What happens if it's absent? 

To explain my confusion further:
`// This allows both the mapping function between two models to be generated (the ModelToModelMappings)`
* Both here indicating two mapping functions? 
* Allows meaning there's further logic down the line that we'll only run if a `ModelToModel` mapping exists?

Reply:
A Field Mapping between a Terraform Schema Model and an SDK Model must have a matching `ModelToModel` (`TerraformModelToModelMappingDefinition`) mapping, so that we know to generate the Mapping functions between the Terraform Schema Model.

As such when we parse the `FieldModelToModel` mappings - that is "The Field X is a model which should be mapped to this other model" - we need to ensure that a `ModelToModel` mapping exists - so that we can output both the call to the mapping function (for the `FieldModelToModel` model) and the mapping function itself (from the `ModelToModel` model).

If the `ModelToModel` mapping doesn't exist but the `FieldModelToModel` mapping does, we'll output a call to a function which doesn't exist - leading to a compilation failure. As such this logic exists both to capture that requirement (whilst we could instead raise an error here, I suspect this'll become the source of truth for that going forwards).

Does that make sense / how's best to explain that one?

Reply:
I think that mostly makes sense, the second paragraph is a little unclear still, is the message still correct if the following adjustments (made in bold italic) are made

>As such when we parse the FieldModelToModel mappings - that is "The **_Field X in a model_** which should be mapped to this other model" - we need to ensure that a ModelToModel mapping exists - so that we can output both the call to the mapping function (for the FieldModelToModel model) and the _**model mapping**_ function itself (from the ModelToModel model).

I think I'm struggling to understand what a "Field which is a model" looks like? and how a field, which for me would be a singular flat property, can be mapped to a model which I view as something that contains multiple properties.